core/fetcher.go

```diff
@@ -0,0 +1,83 @@
+package core
+
+import (
+	"context"
+	"fmt"
+
+	rpctypes "github.com/celestiaorg/celestia-core/rpc/core/types"
+	"github.com/celestiaorg/celestia-core/types"
+	"github.com/celestiaorg/celestia-node/service/block"
+)
+
+const newBlockSubscriber = "NewBlock/Events"
+
+var newBlockEventQuery = types.QueryForEvent(types.EventNewBlock).String()
+
+type BlockFetcher struct {
+	Client
```

Comment:
It should be `client Client`, as BlockFetcher when used should not expose methods of Client. Block Fetcher's responsibility is just to fetch blocks and nothing else.